pkg/columns/formatter/json/json_test.go

```
@@ -141,3 +142,47 @@ func TestDynamicFields(t *testing.T) {
 	formatter := NewFormatter[empty](cols.GetColumnMap())
 	assert.Equal(t, `{"str": "foobar", "int32": 1234567890, "bool": true}`, formatter.FormatEntry(&empty{}))
 }
+
+func TestJSONFormatter(t *testing.T) {
+	type testStruct struct{}
+
+	cols := columns.MustCreateColumns[testStruct]()
+
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child1.grandchild1",
+	}, func(t *testStruct) string { return "parent.child1.grandchild1_text" }))
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child1.grandchild2",
+	}, func(t *testStruct) string { return "parent.child1.grandchild2_text" }))
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child2.grandchild1",
+	}, func(t *testStruct) string { return "parent.child2.grandchild1_text" }))
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child3",
+	}, func(t *testStruct) string { return "parent.child3_text" }))
+
+	assert.NoError(t, cols.AddColumn(columns.Attributes{
+		Name: "parent.child1",
```

IMO this is a programmer issue. Perhaps this should return an error here instead of handling this when serializing the json.